apps/shipments/serializers.py

```diff
@@ -163,10 +174,9 @@ class Meta:
                             'storage_credentials_id', 'contract_version')
 
     def update(self, instance, validated_data):
-        if 'device_id' in validated_data:
+        if 'device_id' in self.context:
             if validated_data['device_id']:
```

Comment:
I think this second `if` is unnecessary.
It could just be :
`if 'device' in self.context:`